### ISSUE TITLE
DLT Pipeline failed during Initialization

### ISSUE THREAD

**Issue by rahulwankhade** [databricks-demos/dbdemos#20]
DLT Pipeline failed during Initialization. Below is the error.

[java.lang.RuntimeException:]([url](url)) Failed to execute python command for notebook '/Users/rahul.wankhade@parle.biz/lakehouse-iot-platform/01-Data-ingestion/01.2-DLT-Wind-Turbine-SQL-UDF' with id RunnableCommandId(5811638086382625936) and error AnsiResult(---------------------------------------------------------------------------
RestException                             Traceback (most recent call last)
File <command--1>:6
      2 import mlflow
      3 #                                                                              Stage/version  
      4 #                                                                 Model name         |        
      5 #                                                                     |              |        
----> 6 predict_churn_udf = mlflow.pyfunc.spark_udf(spark, "models:/dbdemos_turbine_maintenance/Production", "string") #use env_manager='virtualenv' to load the model env (instead of pip install the libs)
      7 spark.udf.register("predict_maintenance", predict_churn_udf)

Kindly let me know what is the issue and how to fix it.

**Comment by QuentinAmbard** [comment]
Hi, I did an update this morning fixing an install issue (0.2.8), and when I run it on my workspace it's working well.
What could happen is that your init job fails & the model isn't created.

When you install the demo we create a job for you. Can you open it & check its status ? If it's failing can you share what's failing in details ?

Thanks
 
![image](https://user-images.githubusercontent.com/96781/227947642-943687d1-3d22-4320-9df3-b713d6d22576.png)

**Comment by rahulwankhade** [comment]
Hi,

Thanks for your prompt response. 
I am using 14 days trial databricks version with premium workspace. I am trying to run DLT pipeline but it fails. Attached is the screenshot & error file for detail explaination.

Also got warning: Magic commands (e.g. %py, %sql and %run) are not supported with the exception of %pip within a Python notebook. Cells containing magic commands are ignored. Unsupported magic commands were found in the following notebooks /Users/rahul.wankhade@parle.biz/lakehouse-iot-platform/01-Data-ingestion/01.1-DLT-Wind-Turbine-SQL:



[](url)
![DatabricksDemosIssue_28 03 2023](https://user-images.githubusercontent.com/15630560/228151066-0a2942c7-16d9-4eff-9323-67bf62125cf8.JPG)
[DatabricksDemos_LoadError.txt](https://github.com/databricks-demos/dbdemos/files/11086187/DatabricksDemos_LoadError.txt)
![DatabricksDemosIssue](https://user-images.githubusercontent.com/15630560/228151107-9c6d338e-97e6-4991-a552-16ac9940f1f9.JPG)

**Comment by jendatx** [comment]
Hi @rahulwankhade I can't tell you why your update failed to run, but I *can* tell you the error is not caused by the magic command warning ("unsupported operation"). I see the same message while mine completes successfully (both the first time and with full-refresh-all). The message is only there because mixed-language notebooks are not currently supported, so the python cell is ignored and the companion source notebook 01.2-DLT-Wind-Turbine-SQL-UDF is python-only and registers the python UDF just fine.

**Comment by rahulwankhade** [comment]
Thanks for your response.

Could you please let me know if any additional configurations requires to run DLT pipeline?
If yes then where parameters needs to be change & which places?

**Comment by QuentinAmbard** [comment]
The error indicates that the model wasn't loaded.
To understand why you need to open the init job. Open the workflow UI, select the job named 
dbdemos_lakehouse_iot_turbine_init_<your_name>
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/96781/228481356-a9b97686-d0a5-44ad-9421-066107b75948.png">
Step 3 register the ML model. Something likely failed during your initialization.
Can you share the error ?

Thanks

**Comment by rahulwankhade** [comment]
Please refer attached error file along with Screenshot.

![DLT_PipelineFails](https://user-images.githubusercontent.com/15630560/228506022-1b2ace1d-4f1d-400e-b336-8f68750ecf10.JPG)
[DLT_PipelineFails_description.txt](https://github.com/databricks-demos/dbdemos/files/11099560/DLT_PipelineFails_description.txt)

**Comment by QuentinAmbard** [comment]
yes so now your model is loaded, if you go back to your DLT pipeline you'll find the full error.
You can click on the last read "start_dlt_pipeline" task, then click on the DLT pipeline link, and then get the message from the DLT run.
It looks like an Azure VM quota

**Comment by tlaur** [comment]
@rahulwankhade I had the same error and it was due to vCPU usage going over the Azure subscription quota limit as pointed out in the message above. You can increase the limits by going to Subscription > Usage + Quotas.
